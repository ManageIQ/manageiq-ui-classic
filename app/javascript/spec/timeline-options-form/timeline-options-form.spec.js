import React from 'react';
import toJson from 'enzyme-to-json';
import fetchMock from 'fetch-mock';
import { shallow } from 'enzyme';
import { act } from 'react-dom/test-utils';
import TimelineOptions from '../../components/timeline-options/timeline-options';
import { sampleReponse, sampleSubmitPressedValues } from './sample-data';

// TODO Snapshots look incorrect

describe('Show Timeline Options form component', () => {
  beforeEach(() => {
    fetchMock
      .once('/api/event_streams', sampleReponse);
  });

  afterEach(() => {
    fetchMock.reset();
    fetchMock.restore();
  });

  // TODO:
  const sampleUrl = `/api/event_streams?limit=5000&offset=0&expand=resources&attributes=group,group_level,group_name,id,event_type,message,ems_id,type,timestamp,created_on&filter[]=type=EmsEvent&filter[]=timestamp>2022-09-07T04:00:00.000Z&filter[]=timestamp<2022-09-28T04:00:00.000Z`;

  /*
 * Render Form
 */

  it('should render form', async(done) => {
    let wrapper;
    await act(async() => {
      wrapper = shallow(<TimelineOptions />);
    });
    setImmediate(() => {
      wrapper.update();
      expect(toJson(wrapper)).toMatchSnapshot();
      done();
    });
  });

  /*
 * Submit Logic
 */

  it('should submit and render table', async(done) => {
    fetchMock.postOnce(sampleUrl, sampleSubmitPressedValues);
    let wrapper;
    await act(async() => {
      wrapper = shallow(<TimelineOptions />);
    });
    expect(toJson(wrapper)).toMatchSnapshot();
    done();
  });
});
