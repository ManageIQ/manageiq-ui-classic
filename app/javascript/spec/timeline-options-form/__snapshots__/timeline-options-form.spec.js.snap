// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Timeline Options form component should render form 1`] = `""`;

exports[`Show Timeline Options form component should submit and render table 1`] = `""`;
